### PR TITLE
Add developer guide framework

### PR DESCRIPTION
**Pull request by canyon289**

### CODE REVIEW

**Comment by OriolAbril** [comment]
Should we merge and iteratively improve it in several PRs?

**Comment by aloctavodia** [comment]
@canyon289 do you want to add more content, or should we merge?

**Comment by canyon289** [comment]
Let me fix up this branch and we can merge. Sorry for staleness

**Comment by canyon289** [comment]
Ok I rebased branch and added some more useful docs. Happy to take comments or merge anytime for iterative development. Thanks for the ping

File: doc/sphinxext/developer_guide.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+following design patterns.
+
+Code Separation
+---------------
+Each backend should be placed in a different module per the backend.
+See `arviz.plots.backends` for examples
+
+The code in the root level of `arviz.plots` should not contain
+any opinion on backend. The idea is that the root level plotting
+function perform math and construct keywords, and the backends
```
Comment (aloctavodia) [inline doc/sphinxext/developer_guide.rst]: typo "functions perform" or "function performs"

File: doc/sphinxext/developer_guide.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+code in `arviz.plots.backends` perform the backend specific
+keyword argument defaulting and plot behavior
+
+The convenience function `get_plotting_function` is available in
```
Comment (aloctavodia) [inline doc/sphinxext/developer_guide.rst]: typo "The convenience function `get_plotting_function` available in"

File: doc/sphinxext/developer_guide.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+`arviz.plots.get_plotting_function` should be called to obtain
+the correct plotting function from the associated backend. If
```
Comment (aloctavodia) [inline doc/sphinxext/developer_guide.rst]: The last sentence is not clear to me.

**Pull request merged**